### DISCUSSION TITLE
AutoControl - Script to enable automation for SelfControl

### DISCUSSION THREAD

**Discussion by seagram** [SelfControlApp/selfcontrol#869] [Show and tell]
Hey everyone 👋,

I've developed an AppleScript that allows you to automate your SelfControl blocking!

Usually, when trying to start a block, SelfControl prompts you to enter your admin password. This step breaks most SelfControl automation techniques.

AutoControl will automatically enter your admin password securely through Keychain Access.
By using AutoControl alongside `cron`, you can set up blocking schedules to automate your SelfControl workflow.

I'd love your feedback for future improvements and features! Hope this helps.

The AutoControl GitHub repository can be found [here](https://github.com/csairlie/AutoControl).

**Comment by slambert** [comment]
Exciting - make sure you let these folks know:

https://github.com/andreasgrill/auto-selfcontrol
https://github.com/Pandoks/hammer-control
